libs/datasets/sources/hhs_hospital_dataset.py

```diff
@@ -1,24 +1,119 @@
+    "hospital_beds_capacity": CommonFields.STAFFED_BEDS,
+    "hospital_beds_in_use": CommonFields.HOSPITAL_BEDS_IN_USE_ANY,
+    "hospital_beds_in_use_covid": CommonFields.CURRENT_HOSPITALIZED,
+}
+
+
+def make_hhs_variable(can_scraper_field, common_field, measurement):
+    """Helper to create a ScraperVariable, since we have a bunch of variables to deal with and
+    two different measurements to deal with (for county and state data)."""
+    return ccd_helpers.ScraperVariable(
+        variable_name=can_scraper_field,
+        measurement=measurement,
+        provider="hhs",
+        unit="beds",
+        common_field=common_field,
+    )
+
+
+# TODO(michael): HHSHospitalStateDataset and HHSHospitalCountyDataset must be separate classes
+# because they both have ScraperVariables that target the same CommonFields. This happens becaues
+# the state level and county level data use the same variable names but different measurements
+# ("current" and "rolling_average_7_day"). This causes CanScraperBase to overwrite itself as it's
+# extracting the variables. Using separate Dataset classes solves this.
+
 
-    COMMON_DF_CSV_PATH = "data/hospital-hhs/timeseries-common.csv"
+class HHSHospitalStateDataset(data_source.CanScraperBase):
+    """Data source for HHS state-level hospital data."""
 
-    EXPECTED_FIELDS = [
-        CommonFields.ICU_BEDS,
-        CommonFields.CURRENT_ICU_TOTAL,
-        CommonFields.CURRENT_ICU,
-        CommonFields.STAFFED_BEDS,
-        CommonFields.HOSPITAL_BEDS_IN_USE_ANY,
-        CommonFields.CURRENT_HOSPITALIZED,
+    SOURCE_TYPE = "HHSHospitalState"
+
+    VARIABLES = [
+        make_hhs_variable(can_field, common_field, "current")
+        for can_field, common_field in FIELD_MAPPING.items()
     ]
 
     @classmethod
     @lru_cache(None)
     def make_dataset(cls) -> MultiRegionDataset:
-        return super().make_dataset().latest_in_static(CommonFields.ICU_BEDS)
+        return modify_dataset(super().make_dataset())
+
+
+class HHSHospitalCountyDataset(data_source.CanScraperBase):
+    """Data source for HHS county-level hospital data."""
+
+    SOURCE_TYPE = "HHSHospitalCounty"
+
+    VARIABLES = [
+        make_hhs_variable(can_field, common_field, "rolling_average_7_day")
+        for can_field, common_field in FIELD_MAPPING.items()
+    ]
+
+    @classmethod
+    @lru_cache(None)
+    def make_dataset(cls) -> MultiRegionDataset:
+        return modify_dataset(super().make_dataset())
+
+
+def modify_dataset(ds: MultiRegionDataset) -> MultiRegionDataset:
+    """Post-processes the can scraper data as necessary."""
+    ts_copy = ds.timeseries.copy()
+
+    # Filter out some of the early noisy data.
+    filtered_ds = dataclasses.replace(
+        ds, timeseries=filter_early_data(ts_copy), timeseries_bucketed=None
+    )
+
+    # Ensure ICU_BEDS is included in the static fields as well.
+    return filtered_ds.latest_in_static(field=CommonFields.ICU_BEDS)
```

Comment:
FWIW- having to explicitly move/copy the latest timeseries data into static data was a surprise to me and wasted a couple hours of building / debugging.  I wonder if CanScraperBase could / should do this automatically somehow?  Or are there cases where we don't want the latest timeseries data automatically promoted to static data?

Reply:
For sure a developer spending hours trying to get a new data source working is a problem. The fix I'm angling for is faster tests that cover the complex parts of our pipeline.
Since there is non-trivial code here a unittest would be nice. You could call `modify_dataset` directly from your test. Or do something like `test_query_source_url` or `test_can_scraper_returns_source_url`.
About this particular issue: long ago we copied the latest timeseries value into the static data somewhere around here but it made the "static" (maybe "scalar" is a better word) data change a lot which made understanding what is happening with it harder. Also we want to get the latest values from timeseries data after the timeseries has gone through all the filtering/aggregation etc.

Reply:
One other point about this is that the whole reason we had this in the first place is because we're getting "static" icu beds from covid care map or the latest value from HHS - since HHS data is complete I don't think we should be using Covid care map at all and we may be able to remove the static data for icu beds

Reply:
@TomGoBravo I don't think either of the tests you suggested would have helped catch this issue, but I can try to add a general dataset test if you want.  Can you point me to a good example to follow?

@ghop02  Yeah.  I was thinking about removing covid care map data as a follow-on PR.

Reply:
A tests where you compare the dataset produced by the hhs class(es) given a small bit of input to an expected dataset could verify that it produced values in `MultiRegionDataset.static`. That doesn't exercise the combined data code that snagged you but does make sure the data source is doing the right thing.
OTOH, if we can remove the dependency on  CovidCareMapBeds then we can stop doing the combined_dataset thing for static hospital/bed capacity values, have the data sources only produce timeseries and let the code downstream pick the latest value. That'd let us remove the call to `latest_in_static` and simplify what is tested.
Argh, so much code for such a simple process :-/

Reply:
https://github.com/covid-projections/covid-data-model/blob/1af9ab5b5a4fa492a238c0e88b4e28ce5bfaff8d/tests/libs/datasets/sources/can_scraper_helpers_test.py#L168-L197
* makes some input data
* defines a class with `_get_covid_county_dataset` overridden to return the input data wrapped in a CanScraperLoader
* calls make_dataset() on the newly defined class
* makes an expected dataset
* compares the expected dataset to the dataset returned by make_dataset.

You want something very similar except the class definition will look more like that at https://github.com/covid-projections/covid-data-model/blob/1af9ab5b5a4fa492a238c0e88b4e28ce5bfaff8d/tests/libs/datasets/data_source_test.py#L51-L54 
```
class CANScraperForTest(HHSHospitalCountyDataset):
        @staticmethod
        def _get_covid_county_dataset():
            return ccd_helpers.CanScraperLoader(input_data)
```
(the method name _get_covid_county_dataset is totally irrelevant today).

Here are two more examples of setting up input data, defining a test class, calling make_dataset on the test class, checking the returned value: https://github.com/covid-projections/covid-data-model/blob/1af9ab5b5a4fa492a238c0e88b4e28ce5bfaff8d/tests/libs/datasets/data_source_test.py#L101-L150

If you aren't already using it, I strongly recommend the debugger in pycharm.

Reply:
Thank you for the pointers, that was immensely helpful.  Sorry I didn't dig harder for these examples myself. I probably should have been able to find them. 😁   Added a couple tests that exercise the date trimming logic as well as validation that we correctly add the data to `latest`.  let me know what you think!